phoenix-core/src/main/java/org/apache/phoenix/mapreduce/PhoenixMultiViewInputFormat.java

```
@@ -0,0 +1,85 @@
+
+public class PhoenixMultiViewInputFormat<T extends Writable> extends InputFormat<NullWritable,T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PhoenixMultiViewInputFormat.class);
+
+    public PhoenixMultiViewInputFormat() {
+    }
+
+    @Override public List<InputSplit> getSplits(JobContext context) {
+        List<InputSplit> listOfInputSplit = new ArrayList<>();
+        try {
+            final Configuration configuration = context.getConfiguration();
+            Class<?> defaultViewTtlDeletionInputStrategyClazz = DefaultPhoenixMultiViewListProvider.class;
+            if (configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_INPUT_STRATEGY_CLAZZ) != null) {
+                defaultViewTtlDeletionInputStrategyClazz = Class.forName(
+                        configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_INPUT_STRATEGY_CLAZZ));
+            }
+            PhoenixMultiViewListProvider phoenixMultiViewListProvider =
+                    (PhoenixMultiViewListProvider) defaultViewTtlDeletionInputStrategyClazz.newInstance();
+            List<ViewInfoWritable> viewsWithTTL = phoenixMultiViewListProvider.getPhoenixMultiViewList(configuration);
+
+            Class<?> defaultViewTtlDeletionSplitStrategyClazz = DefaultMultiViewSplitStrategy.class;
+            if (configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_SPLIT_STRATEGY_CLAZZ) != null) {
+                defaultViewTtlDeletionSplitStrategyClazz = Class.forName(
+                        configuration.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_SPLIT_STRATEGY_CLAZZ));
+            }
+            MultiViewSplitStrategy multiViewSplitStrategy =
+                    (MultiViewSplitStrategy) defaultViewTtlDeletionSplitStrategyClazz.newInstance();
+            listOfInputSplit = multiViewSplitStrategy.generateSplits(viewsWithTTL, configuration);
+        } catch (Exception e) {
+            LOGGER.debug(e.getStackTrace().toString());
```

Don't swallow the exception and log with error.

---

in this block, it might occur `ClassNotFoundException`, `InstantiationException`, and `IllegalAccessException`; however, the base class `InputFormat` doesn't throw a list above. It has 2 options here.
1. wrapper exception to be either `IOException` or `InterruptedException` and throw it.
2. keep current the implementation, which logs exception message.

---

I think we should wrap as an IOException and rethrow